linear_algebra_python/src/lib.py

```diff
@@ -142,7 +144,61 @@ def changeComponent(self,pos,value):
         #precondition
         assert (-len(self.__components) <= pos < len(self.__components))
         self.__components[pos] = value
-    
+    def __eq__(self, other):
+        """
+            eq compares the euclidlength of both vectors.
+        """
+        if not (isinstance(other, Vector)):
+            raise Exception("Invalid Operand")
+        if (len(self)!= len(other)):
+            raise Exception("Vectors do not same number of dimensions")
+        return self.euclidLength()==other.euclidLength()
+    def __lt__(self, other):
+        """
+            lt compares the euclidlength of both vectors.
+        """
+        if not (isinstance(other, Vector)):
+            raise Exception("Invalid Operand")
+        if (len(self)!= len(other)):
+            raise Exception("Vectors do not same number of dimensions")
+        return self.euclidLength()<other.euclidLength()
+    def __le__(self, other):
+        """
+            le compares the euclidlength of both vectors.
+        """
+        if not (isinstance(other, Vector)):
+            raise Exception("Invalid Operand")
+        if (len(self)!= len(other)):
+            raise Exception("Vectors do not same number of dimensions")
+        return self.euclidLength()<=other.euclidLength()         
+    def __gt__(self, other):
+        """
+            gt compares the euclidlength of both vectors.
+        """
+        if not (isinstance(other, Vector)):
+            raise Exception("Invalid Operand")
+        if (len(self)!= len(other)):
+            raise Exception("Vectors do not same number of dimensions")
+        return self.euclidLength()>other.euclidLength()
+    def __ge__(self, other):
+        """
+            ge compares the euclidlength of both vectors.
+        """
+        if not (isinstance(other, Vector)):
+            raise Exception("Invalid Operand")
+        if (len(self)!= len(other)):
+            raise Exception("Vectors do not same number of dimensions")
+        return self.euclidLength()>=other.euclidLength()
+    def __ne__(self, other):
```

Comment:
_Just a suggestion_

Since you have created \_\_eq\_\_ and \_\_lt\_\_ you can just use them in other function _(it will reduce size of the code)_

`<=` `< or ==`
`>` `not <=`
`>=` `> or ==`
`!=` `not ==`

```
def __le__(self, other):
        return self < other or self == other

def __gt__(self, other):
        return not self <= other

def __ge__(self, other):
        return self > other or self == other

def __ne__(self, other):
        return not self == other
```

Reply:
Thx for the insight. I didn't think of that before. Will just make the necessary changes now.